### ISSUE TITLE
`require-returns-check` - async function always returns promise

### ISSUE THREAD

**Issue by tinovyatkin** [gajus/eslint-plugin-jsdoc#174]
So, following is false positive:

```js
/**
 * @returns {Promise}
 */
async foo() {
   await boo();
} 
```

**Comment by gajus** [comment]
PR is welcome.
